src/instrumentsscene/view/abstractinstrumentspaneltreeitem.h

```diff
@@ -66,7 +66,13 @@ class AbstractInstrumentsPanelTreeItem : public QObject
     Q_INVOKABLE virtual bool canAcceptDrop(const QVariant& item) const;
     Q_INVOKABLE virtual void appendNewItem();
 
-    virtual void moveChildren(int sourceRow, int count, AbstractInstrumentsPanelTreeItem* destinationParent, int destinationRow);
+    virtual IDList notationIdsToMove(int sourceRow, int count) const;
```

Comment:
childIdListToMove (or something like that)

Reply:
As far as I can tell, you only need 1 function instead of notationIdsToMove and notationMoveParams:

1. call it buildMoveParams
2. it takes the same arguments as notationMoveParams
3. it will return this structure:

```
struct MoveParams {
    IDList childIdListToMove;
    ID destinationParentId;
    notation::INotationParts::InsertMode insertMode = INotationParts::InsertMode::Before;
}
```